components/vf-hero/vf-hero.scss

```diff
@@ -0,0 +1,154 @@
+// vf-hero
+
+// **Thinking about deleting this file?**
+// If your component needs no CSS/Sass, we still recommend leaving the
+// scss files in place. As this is primarily a CSS framework, it is better to
+// leave the empty files so you know a file wasn't accidently omitted.
+// If you don't have any Sass, you can trim this block down to:
+// "This page was intentionally left blank"
+
+@import 'package.variables.scss';
+// Debug information from component's `package.json`:
+// ---
+/*!
+ * Component: #{map-get($componentInfo, 'name')}
+ * Version: #{map-get($componentInfo, 'version')}
+ * Location: #{map-get($componentInfo, 'location')}
+ */
+
+@import 'vf-hero.variables.scss';
+
+.vf-hero {
+  height: 348px; // needs to be exact height
+  position: relative;
+}
+
+
+
+.vf-hero__content {
+  // only needed when text is centered
+  padding: 1em;
+  position: var(--sticky-position, absolute);
+  width: calc(100% - 28%);
+}
+
+.vf-hero__heading {
+  @include set-type(text-heading--3);
+
+  // positions to the left
+  display: inline-block;
+  margin: 0;
+  padding: .5em;
+}
+
+.vf-hero__text {
+  @include set-type(text-body--2);
+  margin-bottom: 0;
+
+  padding: 1em .5em;
+}
+
+
+.vf-hero {
+  --vf-hero-content-position--inline-start: 0;
+  --vf-hero-content-position--block-end: 0;
+
+  --vf-hero-content-box-shadow: 2px 0 3px rgba(0, 0, 0, var(--vf-hero-content-shadow-alpha, 0)), -2px 0 3px rgba(0, 0, 0, var(--vf-hero-content-shadow-alpha, 0)), 0 2px 3px rgba(0, 0, 0, var(--vf-hero-content-shadow-alpha, 0));
+  --vf-hero-content-box-shadow: 0;
+  --vf-hero__padding: 0;
+
+  --vf-hero-content-shadow-alpha: .1;
+  --vf-hero-heading-bg-color: 1;
+  --vf-hero-heading-text-color: 0;
+
+  --vf-hero-text-bg-color: 0;
+  --vf-hero-text-color: 1;
+}
+
+.vf-hero {
+  margin-bottom: calc( (0 + var(--vf-hero-content-position) ) * 100px );
+}
+
+.vf-hero__content {
+
+  background-color: set-ui-color(vf-ui-color--white);
+  background-color: rgba(calc( (0 + var(--vf-hero-bg-color) ) * 255), calc( (0 + var(--vf-hero-bg-color) ) * 255), calc( (0 + var(--vf-hero-bg-color) ) * 255), calc( (0 + var(--vf-hero-bg-color) ) * 100));
+
+  bottom: -24%;
+  bottom: calc( (0 - var(--vf-hero-content-position--block-end) ) * 1% );
+
+  // box-shadow: 2px 0 3px rgba(0, 0, 0, .1), -2px 0 3px rgba(0, 0, 0, .1), 0 2px 3px rgba(0, 0, 0, .1);
+  box-shadow: var(--vf-hero-content-box-shadow);
+
+  left: 12%;
+  left: calc( (0 + var(--vf-hero-content-position--inline-start) ) * 1% );
+
+  padding: calc( (0 + var(--vf-hero__padding) ) * .1em );
+}
+
+
+.vf-hero__heading {
+  background-color: rgb(calc( (0 + var(--vf-hero-heading-bg-color) ) * 255), calc( (0 + var(--vf-hero-heading-bg-color) ) * 255), calc( (0 + var(--vf-hero-heading-bg-color) ) * 255));
```

Comment:
What's happening here, is this a contrast checking calculation?

Reply:
ah, it's for 'on/off' of black and white. It can probably be a) improved b) documented a little better.